### PR TITLE
[v5][tests]: fix Xcode 10.2, network latency, and TeamCity build issues

### PR DESCRIPTION
**Pull request by mikehardy**
### Summary

CI fails right now and local test-ios fails on crashlytics and firebase.blob problems

CI should go green for ios-e2e with this

I am not sure why Crashlytics headers aren't found during iOS build, but they they aren't.

### Checklist

- [ ] Supports `Android`
- [x] Supports `iOS`
- [x] `e2e` tests added or updated in [/tests/e2e/\*](/tests/e2e)
- [ ] Updated the documentation in the [docs repo](https://github.com/invertase/react-native-firebase-docs)
  - **LINK TO DOCS PR HERE**
- [ ] Flow types updated
- [ ] Typescript types updated

### Test Plan

1. Test Plan
1. Goto 1 (how meta)

### Release Plan

[INTERNAL][BUGFIX] [CI] - skip ios crashlytics, fix races, upgrade deps + lint

---

Think `react-native-firebase` is great? Please consider supporting the project with any of the below:

- 👉 Donate via [Open Collective](https://opencollective.com/react-native-firebase/donate)
- 👉 Follow [`React Native Firebase`](https://twitter.com/rnfirebase) and [`Invertase`](https://twitter.com/invertaseio) on Twitter
- 👉 Star this repo on GitHub ⭐️
- 👉 Contribute; see our [contributing guide](/CONTRIBUTING.md)

### CODE REVIEW

**Comment by mikehardy** [comment]
ci.ios.test went green, and the coverage decrease was sadly expected.
This is a net win though I think, good to merge @Salakar ?

**Comment by mikehardy** [comment]
After getting android working well enough to check this, crashlytics e2e works in Android so I made the e2e disable more surgical - it's only disabled for iOS now, and even though android tests aren't on CI that means when you run them, crashlytics is verified. So that's something.

**Comment by mikehardy** [comment]
CI failed with "can't start build runner", possibly because I force-pushed twice in rapid succession, before the first one was done? https://ci.invertase.io/viewLog.html?buildId=696&buildTypeId=InvertaseOSS_ReactNativeFirebase_CiAnalyse

This all works locally though. Just can't kick the build to check it...

**Comment by mikehardy** [comment]
I gave it time, and force-pushed again and CI proceeds. So just take that as a CI quirk note since I know this CI setup is new

**Comment by mikehardy** [comment]
android suffers intermittent total failure on startup (timeout) but if you quit and retry a few times, it works. So locally now I have ios and android working pretty comfortably. If you can fix the analyze step (some cache issue? prettier configs littered up in the home directory on the filesystem? I dunno) then v5.x.x should be open for business

**Comment by mikehardy** [comment]
What luck, after fixing #2208 I was going after the rest of the dependencies, and it was just picking up a new eslint for some reason. So when I upgraded those packages I reproduced the ci.analyse error locally, fixed it, and ci.analyse works now with the latest commit

**Comment by mikehardy** [comment]
the last commit I pushed just now is only package updates. I tested all of them as I went and they didn't alter coverage stats or break tests or anything, so should be fine? but we can peel that commit off it if breaks the build or  you don't like it

File: tests/e2e/iid/iid.e2e.js [hunk 1]
```diff
@@ -41,8 +41,11 @@ describe('iid()', () => {
 
   describe('deleteToken()', () => {
     it('should return nil from deleteToken with arguments', async () => {
-      const authorizedEntity = firebase.iid().app.options.messagingSenderId;
-      const token = await firebase.iid().deleteToken(authorizedEntity, '*');
+      // This call is racy. On my machine this fails about 40% of the time (over 50 reps)
+      // If I sleep 2 seconds while the token does the auto-background fetch it is stable though
+      await new Promise(resolve => setTimeout(resolve, 10000));
```
Comment (Salakar) [inline tests/e2e/iid/iid.e2e.js]:
```suggestion
      await sleep(10000);
```

File: tests/e2e/messaging/messaging.e2e.js [hunk 2]
```diff
@@ -127,10 +127,18 @@ describe('messaging()', () => {
...
+      // If I sleep 2 seconds while the token does the auto-background fetch it is stable though
+      await new Promise(resolve => setTimeout(resolve, 10000));
+
       const tokenBefore = await firebase.messaging().getToken();
       tokenBefore.should.be.a.String();
       await firebase.messaging().deleteToken();
 
+      // This call is racy. On my machine this fails about 40% of the time (over 50 reps)
+      // If I sleep 2 seconds while the token does the auto-background fetch it is stable though
+      await new Promise(resolve => setTimeout(resolve, 10000));
```
Comment (Salakar) [inline tests/e2e/messaging/messaging.e2e.js]:
```suggestion
      await sleep(10000);
```

File: tests/e2e/messaging/messaging.e2e.js [hunk 3]
```diff
@@ -127,10 +127,18 @@ describe('messaging()', () => {
 
   describe('deleteToken()', () => {
     it('deletes the current fcm token', async () => {
+      // This call is racy. On my machine this fails about 40% of the time (over 50 reps)
+      // If I sleep 2 seconds while the token does the auto-background fetch it is stable though
+      await new Promise(resolve => setTimeout(resolve, 10000));
```
Comment (Salakar) [inline tests/e2e/messaging/messaging.e2e.js]:
```suggestion
      await sleep(10000);
```

**Comment by Salakar** [comment]
This is awesome, thanks Mike! I made some suggestions (minor) - would commit these myself to the PR but I don't have access:

image

----

I'm happy to merge either way though as these aren't that important. I'm going to work on the remaining Android CI bits over the weekend for v6 and I'll backport this to v5 so it's also tested on CI - hopefully helps.

**Comment by Salakar** [comment]
Two approvals - feel free to merge when you want to. 👍

**Comment by mikehardy** [comment]
> This is awesome, thanks Mike! I made some suggestions (minor) - would commit these myself to the PR but I don't have access:
> 
> image

That's odd, I always leave "Allow edits from maintainers" checked? Either way, yeah didn't realize you had a sleep function already, no need to sprinkly the one-liner everywhere when I can call the util

> 
> I'm happy to merge either way though as these aren't that important. I'm going to work on the remaining Android CI bits over the weekend for v6 and I'll backport this to v5 so it's also tested on CI - hopefully helps.

I've tried to keep the commit history squeaky clean so it is rebase-mergeable vs squash-mergeable and I like being able to bisect so I prefer multiple commits, but I know some repos have opinions on "never rebase" or "never squash". In this case I'd like to rebase merge vs squash but if you guys philosophize differently I'll defer to your style - just let me know. Otherwise rebase it'll be

**Comment by Salakar** [comment]
Rebase and merge now enabled, go for it

**Comment by mikehardy** [comment]
responding to the comments with a new commit re-requires approvals :-) - this one is ready now though. v5 is green local android + ios with cover, green ios on CI, and has a clean `yarn outdated` bill of health in main and tests woot!

**Pull request merged**